lib/core/asynctree.js

```diff
@@ -129,18 +129,18 @@ class AsyncTree extends EventEmitter{
         err.stack = err.stack.split('\n').slice(1).join('\n');
       }
 
-      if (this.shouldRejectNodePromise(err)) {
-        this.currentNode.reject(err);
+      if (this.shouldRejectNodePromise(err, node)) {
+        node.reject(err);
       } else {
-        this.currentNode.resolve(err);
+        node.resolve(err);
       }
 
-      if (this.shouldRejectParentNodePromise(err)) {
+      if (this.shouldRejectParentNodePromise(err, node)) {
         parent.reject(err);
       }
     } else {
       node.resolveValue = result;
-      this.resolveNode(this.currentNode, result);
+      this.resolveNode(node, result);
```

Comment:
Instead of using `this.currentNode` which might have changed during execution (When there are events like abortOnFailure `currentNode` is shifted to `rootNode`), I've changed it `node` which is stored in the context while executing this function.